### PR TITLE
NAM contract type for ACTUS

### PR DESCRIPTION
**Pull request by cheezus1**
This PR:

- Introduces NAM contract type for ACTUS (POF, SCHED, STF, State initialization)
- Adds applicability rules for NAM and improves the ones for PAM and LAM contract types
- Refactors year fraction function so that it takes `Maybe Day` since the MD contract term is optional
- Removes the Marlowe fs and contract terms files written in tests
- Adds NAM contract type to Marlowe playground

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

Pre-merge checklist:
- [x] Someone approved it
- [x] Commits have useful messages
- [x] Review clarifications made it into the code
- [x] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: marlowe-actus/src/Language/Marlowe/ACTUS/Model/Utility/YearFraction.hs [hunk 1]
```diff
@@ -40,6 +42,8 @@ yearFraction DCC_A_365 startDay endDay _
   = 0.0
 
 yearFraction DCC_E30_360ISDA startDay endDay maturityDate
+  | isNothing maturityDate
+  = error $ "DCC_E30_360ISDA requires maturity date"
```
Comment (palas) [inline marlowe-actus/src/Language/Marlowe/ACTUS/Model/Utility/YearFraction.hs:46]: In this case, we could match directly in the clauses:
```haskell
yearFraction DCC_E30_360ISDA _ _ Nothing = error "DCC_E30_360ISDA requires maturity date"
yearFraction DCC_E30_360ISDA startDay endDay (Just maturityDate)
  | startDay <= endDay
  = let
  ...
```
And then we wouldn't need the `fromJust` in line 55

File: marlowe-actus/test/Spec/Marlowe/Actus.hs [hunk 2]
```diff
@@ -108,13 +129,32 @@ pamStatic = case genStaticContract contractTerms of
...
+namProjected = do
+    let cfs = genProjectedCashflows namContractTerms
+    let cfsEmpty = null cfs
+    assertBool "Cashflows should not be empty" (not cfsEmpty)
 
+namStatic :: IO ()
+namStatic = case genStaticContract namContractTerms of
+  Failure _ -> assertFailure "Terms validation should not fail"
+  Success contract ->
+    do
```
Comment (palas) [inline marlowe-actus/test/Spec/Marlowe/Actus.hs:149]: I think this `do` is not necessary

File: marlowe-actus/src/Language/Marlowe/ACTUS/Model/STF/StateTransitionModel.hs [hunk 3]
```diff
@@ -231,3 +231,51 @@ _STF_SC_LAM st@ContractStatePoly{..} t y_sd_t y_tfpminus_t y_tfpminus_tfpplus _F
...
+_STF_RR_NAM st t y_sd_t y_tfpminus_t y_tfpminus_tfpplus _FEB _FER _CNTRL _RRLF _RRLC _RRPC _RRPF _RRMLT _RRSP o_rf_RRMO =
+    _STF_RR_LAM st t y_sd_t y_tfpminus_t y_tfpminus_tfpplus _FEB _FER _CNTRL _RRLF _RRLC _RRPC _RRPF _RRMLT _RRSP o_rf_RRMO
+
+_STF_RRF_NAM st t y_sd_t y_tfpminus_t y_tfpminus_tfpplus _FEB _FER _CNTRL _RRNXT =
+    _STF_RRF_LAM st t y_sd_t y_tfpminus_t y_tfpminus_tfpplus _FEB _FER _CNTRL _RRNXT
+
+_STF_SC_NAM st t y_sd_t y_tfpminus_t y_tfpminus_tfpplus _FEB _FER _CNTRL _SCEF o_rf_SCMO _SCIED =
+    _STF_SC_LAM st t y_sd_t y_tfpminus_t y_tfpminus_tfpplus _FEB _FER _CNTRL _SCEF o_rf_SCMO _SCIED
+
+_STF_CE_NAM st t y_sd_t = _STF_AD_PAM st t y_sd_t
```
Comment (palas) [inline marlowe-actus/src/Language/Marlowe/ACTUS/Model/STF/StateTransitionModel.hs:281]: I have noticed that this `_STF_CE_NAM` points to `_STF_AD_PAM`  in several places. Just wanted to check whether this is on purpose

**Review by palas — APPROVED** [review]
Looks good to me. Just a couple of suggestions

File: marlowe-actus/src/Language/Marlowe/ACTUS/Model/APPLICABILITY/ApplicabilityModel.hs [hunk 4]
```diff
@@ -4,38 +4,82 @@
...
+_X_I_1 _uncondCts _condCt t@ContractTerms{..} uncondNames condName =
+    case any id _uncondCts of
+        True ->
+            Success t
+        False ->
+            case _condCt t of
+                Just _ ->
+                    Failure [Required $ "The unconditional terms " ++ show uncondNames ++ " must be defined when " ++ condName ++ " is defined for contract type '" ++ show (fromJust contractType) ++ "'"]
+                Nothing ->
+                    Success t
```
[the remark this answers is not in the document]
Comment (cheezus1) [reply marlowe-actus/src/Language/Marlowe/ACTUS/Model/APPLICABILITY/ApplicabilityModel.hs:27]: For now, we just wrote our code, following the established style. Refactoring will be done in another PR.

File: marlowe-actus/src/Language/Marlowe/ACTUS/Model/STF/StateTransitionModel.hs (hunk 3 above)
Comment (cheezus1) [reply marlowe-actus/src/Language/Marlowe/ACTUS/Model/STF/StateTransitionModel.hs:281]: It is done on purpose, yes, as stated in the ACTUS specs.

**Pull request merged**